sisyphus/engine.py

```diff
@@ -201,6 +202,18 @@ def init_worker(self, task):
         """
         pass
 
+    def get_job_node_hostnames(self) -> List[str]:
```

Comment:
would you implement this function also in `EngineSelector` and the other engines we have implemented?